src/sdk/pynni/nni/bohb_advisor/bohb_advisor.py

```diff
@@ -0,0 +1,587 @@
+    min_budget: float
+		The smallest budget to consider. Needs to be positive!
+	max_budget: float
+		The largest budget to consider. Needs to be larger than min_budget!
+		The budgets will be geometrically distributed
+        :math:`a^2 + b^2 = c^2 \sim \eta^k` for :math:`k\in [0, 1, ... , num\_subsets - 1]`.
+    eta: int
+		In each iteration, a complete run of sequential halving is executed. In it,
+		after evaluating each configuration on the same subset size, only a fraction of
+		1/eta of them 'advances' to the next round.
+		Must be greater or equal to 2.
+    min_points_in_model: int
+		number of observations to start building a KDE. Default 'None' means
+		dim+1, the bare minimum.
+    top_n_percent: int
+		percentage ( between 1 and 99, default 15) of the observations that are considered good.
+	num_samples: int
+		number of samples to optimize EI (default 64)
+	random_fraction: float
+		fraction of purely random configurations that are sampled from the
+		prior without the model.
+	bandwidth_factor: float
+		to encourage diversity, the points proposed to optimize EI, are sampled
+		from a 'widened' KDE where the bandwidth is multiplied by this factor (default: 3)
+	min_bandwidth: float
+		to keep diversity, even when all (good) samples have the same value for one of the parameters,
+		a minimum bandwidth (Default: 1e-3) is used instead of zero.
+    """
+    def __init__(self,
+                 optimize_mode='maximize',
+                 min_budget=1,
+                 max_budget=3,
+                 eta=3,
+                 min_points_in_model=None,
+                 top_n_percent=15,
+                 num_samples=64,
+                 random_fraction=1/3,
+                 bandwidth_factor=3,
+                 min_bandwidth=1e-3):
+        super()
+        self.optimize_mode = OptimizeMode(optimize_mode)
+        self.min_budget = min_budget
+        self.max_budget = max_budget
+        self.eta = eta
+        self.min_points_in_model = min_points_in_model
+        self.top_n_percent = top_n_percent
+        self.num_samples = num_samples
+        self.random_fraction = random_fraction
+        self.bandwidth_factor = bandwidth_factor
+        self.min_bandwidth = min_bandwidth
+
+        # all the configs waiting for run
+        self.generated_hyper_configs = []
+        # all the completed configs
+        self.completed_hyper_configs = []
+
+        self.s_max = math.floor(
+            math.log(self.max_budget / self.min_budget, self.eta) + _epsilon)
+        # current bracket(s) number
+        self.curr_s = self.s_max
+        # In this case, tuner increases self.credit to issue a trial config sometime later.
+        self.credit = 0
+        self.brackets = dict()
+        self.search_space = None
+        # [key, value] = [parameter_id, parameter]
+        self.parameters = dict()
+
+        # config generator
+        self.cg = None
+
+    def load_checkpoint(self):
+        pass
+
+    def save_checkpont(self):
```

Comment:
save_checkpoint

Reply:
This function is inherited from "MsgDispatcherBase". Currently, we don't have this feature.

Reply:
I mean the correct spelling of "save_checkpoint" not "save_checkpont". I think you can help correcting the name in "recoverable.py" and "hyperband_advisor.py".
XD

Reply:
I see. thx, will change the in "recoverable.py" and "hyperband_advisor.py" in #pr911